exampleCourse/questions/element/dataframe/server.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import prairielearn as pl
+import pandas as pd
+
+
+def generate(data):
+    df = pd.io.parsers.read_csv("breast-cancer-train.dat", header=None)
+
+    df2 = pd.DataFrame(
+        [
+            {
+                "city": "Champaign",
+                "job": "Professor",
+                "age": 35,
+                "time": pd.to_datetime("2022-10-06 12:00"),
+            },
+            {
+                "city": "Sunnyvale",
+                "job": "Driver",
+                "age": 20,
+                "time": pd.to_datetime("2020-05-09 12:00"),
+            },
+            {
+                "city": "Mountain View",
+                "job": "Data Scientist",
+                "age": np.nan,
+                "time": pd.to_datetime("2021-12-14 12:00"),
+            },
+        ]
+    )
+
+    dft = pd.DataFrame(
+        {
+            # Scalars
+            "integer": 1,
+            "numeric": 3.14,
+            "logical": False,
+            "character": "foo",
+            # TODO adding in complex numbers won't deserialize correctly, fix this (somehow?)
+            # "complex": complex(1, 2),
```

Comment:
@coatless do you know why the complex numbers won't reserialize correctly? It looks like the output has a dict with real and imaginary parts, but this isn't correctly converted back during the deserialization step (just calls float constructor on this dict).

Reply:
@eliotwrobson thanks for the heads up. It looks like Pandas' complex number support has a few hiccups. 

I've raised the issue upstream at: https://github.com/pandas-dev/pandas/issues/50782

That said, I'm not sure if this is a show-stopper or not. In particular, the original `dataframe-v1` is able to be _reconstructed_ with complex numbers but lacks `missing value`/`time` support. Over time, I'm sure that issue can get resolved.

Some experimentations with complex numbers: 

https://colab.research.google.com/github/coatless/colab-notes/blob/main/03-dataframe-json-export-and-import-with-missing-and-date-values-prairielearn.ipynb#scrollTo=lqZFltSTHsAx

Reply:
I don't think this is a show stopper then. If people need complex numbers they can use the old encoding, and we'll pick up any fixes that get deployed in the main Pandas repo too.

Reply:
Aye, that was my thought as well.